### PR TITLE
feature CSI 2789 support empty string SpaceEfficiency

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

**Review by oriyarde — DISMISSED** [review]
LGTM

File: controller/tests/controller_server/csi_controller_server_test.py [hunk 1]
```diff
@@ -384,10 +384,22 @@ def test_create_volume_succeeds(self, storage_agent):
...
         self.mediator.get_volume.assert_called_once_with(volume_name, pool_id='pool1')
-        self.mediator.create_volume.assert_called_once_with(volume_name, 10, {}, 'pool1')
+        self.mediator.create_volume.assert_called_once_with(volume_name, 10, None, 'pool1')
         self.assertEqual(response_volume.volume.content_source.volume.volume_id, '')
         self.assertEqual(response_volume.volume.content_source.snapshot.snapshot_id, '')
 
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_create_volume_with_space_efficiency_succeeds(self, storage_agent):
+        self._prepare_create_volume_mocks(storage_agent)
+        self.request.parameters.update({config.PARAMETERS_SPACE_EFFICIENCY: "not_none"})
```
Comment (oriyarde) [inline controller/tests/controller_server/csi_controller_server_test.py:394]: where is the unit test with the empty string?

Comment (ArbelNathan) [reply controller/tests/controller_server/csi_controller_server_test.py:394]: its not intersting in `controller_server`, its not handling the values

Comment (oriyarde) [reply controller/tests/controller_server/csi_controller_server_test.py:394]: so where is it in svc mediator?

Comment (ArbelNathan) [reply controller/tests/controller_server/csi_controller_server_test.py:394]: [here](https://github.com/IBM/ibm-block-csi-driver/blob/41fd19583216b33b1eab3614fbb7a3301b876a3b/controller/tests/array_action/svc/array_mediator_svc_test.py#L468)

Comment (oriyarde) [reply controller/tests/controller_server/csi_controller_server_test.py:394]: I meant a "create volume" unit test

Comment (ArbelNathan) [reply controller/tests/controller_server/csi_controller_server_test.py:394]: you right,  added for all se types

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by zingero — APPROVED** [review]

**Pull request merged**